src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs

```diff
@@ -106,9 +106,9 @@ public abstract class UntypedObjectBase : IUntypedObject
         /// <summary>
         /// Set a property on the object.
         /// </summary>
-        /// <param name="propertyName">Property name.</param>
+        /// <param name="property">Property identifier. Usually a number or name.</param>

```

Comment:
>Property identifier. Usually a number or name. [](http://example.com/codeflow?start=35&length=46)

Why can't we keep string and use "1" when it's a number?

Reply:
just add a new virtual. 
SetProperty(string ) 
SetIndex(int i) 

This simplifies responsibly on the host that has to implement these.  We don't want every host having to check for both cases.